src/Codegen/Builders/ObjectBuilder.hack

```diff
@@ -58,7 +58,7 @@ final class ObjectBuilder<TField as IFieldBuilder> extends OutputTypeBuilder<\Sl
                 },
             )
             ->addDefault()
-            ->addLine("throw new \Exception('Unknown field: '.\$field_name);")
+            ->addLine("return null;")
```

Comment:
Moving this into the implementation of `resolveAsync` so that we can use `getFieldDefinition` to determine whether a field is valid on a parent.